banzai_floyds/utils/telluric_utils.py

```diff
@@ -14,4 +26,80 @@
-                    {'wavelength_min': 8730.0, 'wavelength_max': 9800.0, 'molecule': 'H2O'}]
+                    {'wavelength_min': 8730.0, 'wavelength_max': 9960.0, 'molecule': 'H2O'},
+                    {'wavelength_min': 10500.0, 'wavelength_max': 12500.0, 'molecule': 'H2O'}]
+
+
+def scale_trasmission(transmission, airmass_scale):
+    # This was a giant pain to figure out.
+    # We assume the transmission is an exponential with airmass ala the Beer-Lambert Law
+    # T = e^(-A(λ) * X) where X is the airmass and A(λ) is the absorption for a given atmospheric composition
+    # If we consider the transmission for some airmass, X1
+    # ln(T1) = -A(λ) * X1
+    # A(λ) = -log(λ) / X1
+    # T2 = e^(log(T1) / X1 * X2)
+    # Using identities of exponents,
+    # T2 = T1^(X2 / X1)
+    return transmission ** airmass_scale
+
+
+def get_molecular_regions(wavelength):
+    o2_region = np.zeros_like(wavelength, dtype=bool)
+    h2o_region = np.zeros_like(wavelength, dtype=bool)
+    for region in TELLURIC_REGIONS:
+        telluric_wavelengths = np.logical_and(wavelength >= region['wavelength_min'],
+                                              wavelength <= region['wavelength_max'])
+        if region['molecule'] == 'O2':
+            o2_region = np.logical_or(o2_region, telluric_wavelengths)
+        elif region['molecule'] == 'H2O':
+            h2o_region = np.logical_or(h2o_region, telluric_wavelengths)
+    return o2_region, h2o_region
+
+
+def scale_telluric(telluric_transmission, wavelength, o2_scale, h2o_scale, h2o_region=None, o2_region=None):
+    scaled_model = telluric_transmission.copy()
+    if o2_region is None or h2o_region is None:
+        o2_region, h2o_region = get_molecular_regions(wavelength)
+    scaled_model[h2o_region] = scale_trasmission(scaled_model[h2o_region], h2o_scale)
+    scaled_model[o2_region] = scale_trasmission(scaled_model[o2_region], o2_scale)
+    return {'telluric': scaled_model, 'wavelength': wavelength}
+
+
+def estimate_telluric(wavelength, airmass, elevation, telluric_model=None, resolution_fwhm=17.5):
+    if telluric_model is None:
+        # Load the default telluric absorption model from Matheson 2000
+        telluric_model = ascii.read(pkg_resources.resource_filename('banzai_floyds', 'data/telluric.dat'))
+        # Convolve with the resolution of FLOYDS
+        # The default value of 5 pixels is picked for the 2" slit. This only needs to be roughly correct.
+        # The telluric model from Matheson 2000 is sampled at 1 Angstrom per pixel so use a resolution of 17.5
+        sigma = fwhm_to_sigma(resolution_fwhm)
+        telluric_model['telluric'] = gaussian_filter1d(telluric_model['telluric'], sigma, mode='constant', cval=1.0)
+        # We adopt the elevation of KPNO for which this was measured on the McMath-Pierce Solar Telescope
+        # Don't rescale the telluric model at all for now
+        # airmass_ratio = elevation_to_airmass_ratio(elevation, 2096)
```

Comment:
A comment about all of these commented lines might be nice.
Why are these lines here? Should they be removed?